sorts/binary_insertion_sort.py

```diff
@@ -0,0 +1,96 @@
+"""
+A pure Python implementation of the binary insertion sort algorithm
+
+Binary insertion sort is a sorting algorithm which is similar to the insertion sort,
+but instead of using linear search to find the location where an element should be
+inserted, we use binary search. Thus, we reduce the comparative value of inserting
+a single element from O (N) to O (log N).
+
+For doctests run following command:
+python3 -m doctest -v binary_insertion_sort.py
+
+For manual testing run:
+python3 binary_insertion_sort.py
+"""
+
+
+def binary_search(collection: list[int], value: int, start: int, end: int) -> int:
```

Comment:
This algorithm is incorrect and breaks with the following initial tests:
```python
binary_search([1, 2, 3, 4, 5], 2, 0, 1)  # gives 2
binary_search([1, 2, 3, 4, 5], 2, 0, 4)  # gives 2
```

Please test out the algorithms thoroughly. Also, if possible, use existing implementation from `searches/binary_search.py`